File: power-of-two/powerOfTwo.go
package main

import "fmt"

func isPowerOfTwo(n int) bool {
	if n <= 0 {
		return false
	} else if n == 1 {
		return true
	}
	for n%2 == 0 {
		n /= 2
	}
	return n == 1
}

func isPowerOfTwoBit(n int) bool {
	if n <= 0 {
		return false
	}
	return n&(n-1) == 0
}

func main() {
	fmt.Println(isPowerOfTwo(1))
	fmt.Println(isPowerOfTwo(16))
	fmt.Println(isPowerOfTwo(218))
	fmt.Println(isPowerOfTwoBit(1))
	fmt.Println(isPowerOfTwoBit(16))
	fmt.Println(isPowerOfTwoBit(218))
}

File: power-of-two/powerOfTwo.js
//Problem : Determine given a positive integer "n", determine if the number is a power of two or not.

function isPowerOfTwo(num) {
    if (num === 1) {
        return true;
    }
    else if (num % 2 === 0) {
        return isPowerOfTwo(num / 2);
    } else {
        return false;
    }
}

function isPowerOfTwoBit(num) {
    if (num === 1) {
        return true;
    }
    return (num & (num - 1)) === 0;
}

let num = Number(process.argv[2]);
console.log(isPowerOfTwo(num));
console.log(isPowerOfTwoBit(num));